test/components/views/context_menus/MessageContextMenu-test.tsx

```diff
@@ -13,35 +13,220 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 
 import React from 'react';
-import { mount } from 'enzyme';
+import { mount, ReactWrapper } from 'enzyme';
 import { EventStatus, MatrixEvent } from 'matrix-js-sdk/src/models/event';
 import { Room } from 'matrix-js-sdk/src/models/room';
 import { PendingEventOrdering } from 'matrix-js-sdk/src/matrix';
 import { ExtensibleEvent, MessageEvent, M_POLL_KIND_DISCLOSED, PollStartEvent } from 'matrix-events-sdk';
+import { Thread } from "matrix-js-sdk/src/models/thread";
 
 import * as TestUtils from '../../../test-utils';
-import MessageContextMenu from '../../../../src/components/views/context_menus/MessageContextMenu';
 import { MatrixClientPeg } from '../../../../src/MatrixClientPeg';
+import RoomContext, { TimelineRenderingType } from "../../../../src/contexts/RoomContext";
+
+const PATH_TO_STRING_UTILS = "../../../../src/utils/strings";
+const PATH_TO_EVENT_UTILS = "../../../../src/utils/EventUtils";
+
+jest.mock(PATH_TO_STRING_UTILS);
+jest.mock(PATH_TO_EVENT_UTILS);
+
+const { copyPlaintext, getSelectedText } = require(PATH_TO_STRING_UTILS);
+const { canEditContent, canForward, isContentActionable } = require(PATH_TO_EVENT_UTILS);
+
+describe('MessageContextMenu', () => {
+    beforeAll(() => {
+        jest.resetAllMocks();
+    });
 
-describe('MessageContextMenu>', () => {
     it('allows forwarding a room message', () => {
+        canForward.mockReturnValue(true);
+        isContentActionable.mockReturnValue(true);
+
         const eventContent = MessageEvent.from("hello");
-        const menu = createMessageContextMenu(eventContent);
+        const menu = createMenuWithContent(eventContent);
         expect(menu.find('div[aria-label="Forward"]')).toHaveLength(1);
     });
 
     it('does not allow forwarding a poll', () => {
+        canForward.mockReturnValue(false);
+
         const eventContent = PollStartEvent.from("why?", ["42"], M_POLL_KIND_DISCLOSED);
-        const menu = createMessageContextMenu(eventContent);
+        const menu = createMenuWithContent(eventContent);
         expect(menu.find('div[aria-label="Forward"]')).toHaveLength(0);
     });
+
+    it('does show copy link button when supplied a link', () => {
+        const eventContent = MessageEvent.from("hello");
+        const props = {
+            link: "https://google.com/",
+        };
+        const menu = createMenuWithContent(eventContent, props);
+        const copyLinkButton = menu.find('a[aria-label="Copy link"]');
+        expect(copyLinkButton).toHaveLength(1);
+        expect((copyLinkButton.getDOMNode() as HTMLAnchorElement)?.href).toBe(props.link);
+    });
+
+    it('does not show copy link button when not supplied a link', () => {
+        const eventContent = MessageEvent.from("hello");
+        const menu = createMenuWithContent(eventContent);
+        const copyLinkButton = menu.find('a[aria-label="Copy link"]');
+        expect(copyLinkButton).toHaveLength(0);
+    });
+
+    it('(right click) copy button does work as expected', () => {
```

Comment:
All of these `(right click)` tests could be nested inside a `describe("right click variant", () => …` block to avoid repeating that label